gemma-core/src/main/java/ubic/gemma/core/ontology/OntologyService.java

```diff
@@ -101,6 +101,12 @@ Collection<CharacteristicValueObject> findExperimentsCharacteristicTags( String
      */
     Collection<OntologyTerm> getCategoryTerms();
 
+    /**
+     *
+     * @return terms allowed for the predicate (relationship) in a Characteristic
+     */
+    Collection<OntologyTerm> getRelationTerms();
+
```

Comment:
This should be a collection of `OntologyProperty`. There's two possible subclasses: `ObjectProperty` and `DatatypeProperty`. We can use the latter for free-text terms.